spark/src/main/scala/org/apache/spark/sql/delta/DeltaColumnMapping.scala

```diff
@@ -156,10 +153,6 @@ trait DeltaColumnMappingBase extends DeltaLogging {
             (newProtocol.supportsWriterFeatures && satisfiesReaderVersion)) {
           newProtocol = newProtocol.withFeature(ColumnMappingTableFeature)
         }
-
-        if (!satisfiesColumnMappingProtocol(newProtocol)) {
-          throw DeltaErrors.changeColumnMappingModeOnOldProtocol(oldProtocol)
-        }
```

Comment:
If you're deleting this code, you can actually delete everything under the `else` here and simplify this whole method. It's a lot of throwaway work :D

Reply:
Right of course.